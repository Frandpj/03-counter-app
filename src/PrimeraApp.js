import React from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';

// FC (Funtional components)

const PrimeraApp = ({ saludo, subtitulo }) => {

    // const saludo = 'Hola Mundo';

    return (
        <>
            <h1>{ saludo }</h1>
            {/* <pre>{ JSON.stringify(saludo, null, 3) }</pre> */}
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo',
}

export default PrimeraApp;