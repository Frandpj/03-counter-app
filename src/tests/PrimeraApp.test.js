import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp />', () => {
    
    // test('Debe de mostrar el mensaje "Hola soy Francisco"', () => {
   
    //     const saludo = 'Hola soy Francisco';

    //     // Desestructuraci√≥n
    //     const {getByText} = render(<PrimeraApp saludo = {saludo}/>)

    //     expect(getByText(saludo)).toBeInTheDocument();

    // });

    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
        
        const saludo = "Hola soy Francisco";
        const wrapper = shallow(<PrimeraApp saludo = {saludo}/>);

        // Crea un snapshot de lo que enviamos, en este caso el componente completo
        // <PrimeraApp />
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar el subtitulo enviado por props', () => {

        const saludo = "Hola soy Francisco";
        const subtitulo = "Soy un subtitulo";

        const wrapper = shallow(
            <PrimeraApp 
                saludo = {saludo}
                subtitulo = {subtitulo}
            />
        );
       
        const textoParrafo = wrapper.find('p').text();
        
        expect(textoParrafo).toBe(subtitulo);
        
    });
    
    
    

});
