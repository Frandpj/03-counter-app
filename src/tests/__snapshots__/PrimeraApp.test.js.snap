// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraAopp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Francisco
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
